storage/rocksdb/ha_rocksdb.cc

```diff
@@ -12443,16 +12885,203 @@ void Rdb_background_thread::run() {
   ddl_manager.persist_stats();
 }
 
+void Rdb_index_stats_thread::run() {
+  const int WAKE_UP_INTERVAL = 1;
+#ifdef TARGET_OS_LINUX
+  RDB_MUTEX_LOCK_CHECK(m_is_mutex);
+  m_tid_set = true;
+  m_tid = syscall(SYS_gettid);
+  RDB_MUTEX_UNLOCK_CHECK(m_is_mutex);
+#endif
+
+  renice(rocksdb_table_stats_background_thread_nice_value);
+  for (;;) {
+    RDB_MUTEX_LOCK_CHECK(m_signal_mutex);
+    if (m_killed) {
+      RDB_MUTEX_UNLOCK_CHECK(m_signal_mutex);
+      break;
+    }
+
+    timespec ts;
+    clock_gettime(CLOCK_REALTIME, &ts);
+
+    // Wait for 24 hours if the table scan based index calculation
```

Comment:
Stalling a background/worker thread for 24 hours?  This doesn't look like a proper design.  We should look closer at this before taking in this commit.

Reply:
Everything is fine here. When `rocksdb_table_stats_use_table_scan` is switched to `ON` we have the following calls that will unblock `mysql_cond_timedwait(&m_signal_cond, &m_signal_mutex, &ts);`:
```
ha_rocksdb::write_row
ha_rocksdb::update_table_stats_if_needed
Rdb_index_stats_thread::add_index_stats_request
Rdb_index_stats_thread::signal
```
`Rdb_index_stats_thread::signal` is called also with `rocksdb_done_func()`.

Reply:
Basically it causes the table scan based index calculation every 24 hours when `rocksdb_table_stats_use_table_scan` is `OFF`.

Reply:
OK, makes sense.